Source/WebCore/platform/ios/PlatformScreenIOS.mm

```diff
@@ -129,6 +129,9 @@ FloatRect screenAvailableRect(Widget* widget)
 
 float screenPPIFactor()
 {
+    if (auto data = screenData(primaryScreenDisplayID()))
```

Comment:
Can we just ensure that we always have screenData instead of having two code paths?

Reply:
> Can we just ensure that we always have screenData instead of having two code paths?

Yes, that is a good point. Currently, the screenPPIFactor function is called from WebCore, which does not have knowledge about which process it is being called from. I wonder if we should send the screen data to all WebKit processes to make sure there always is a cached version? We would also need to cache the screen data in WK1.

Thanks for reviewing!